homeassistant/components/arve/sensor.py

```diff
@@ -0,0 +1,141 @@
+        native_unit_of_measurement=CONCENTRATION_PARTS_PER_MILLION,
+        device_class=SensorDeviceClass.CO2,
+        value_fn=lambda arve_data: getattr(arve_data, "co2"),
+    ),
+    ArveDeviceEntityDescription(
+        key="AQI",
+        translation_key="aqi",
+        native_unit_of_measurement=None,
+        device_class=SensorDeviceClass.AQI,
+        value_fn=lambda arve_data: getattr(arve_data, "aqi"),
+    ),
+    ArveDeviceEntityDescription(
+        key="Humidity",
+        translation_key="humidity",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.HUMIDITY,
+        value_fn=lambda arve_data: getattr(arve_data, "humidity"),
+    ),
+    ArveDeviceEntityDescription(
+        key="PM10",
+        translation_key="pm10",
+        native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+        device_class=SensorDeviceClass.PM10,
+        value_fn=lambda arve_data: getattr(arve_data, "pm10"),
+    ),
+    ArveDeviceEntityDescription(
+        key="PM25",
+        translation_key="pm25",
+        native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+        device_class=SensorDeviceClass.PM25,
+        value_fn=lambda arve_data: getattr(arve_data, "pm25"),
+    ),
+    ArveDeviceEntityDescription(
+        key="Temperature",
+        translation_key="temperature",
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        value_fn=lambda arve_data: getattr(arve_data, "temperature"),
+    ),
+    ArveDeviceEntityDescription(
+        key="TVOC",
+        translation_key="tvoc",
+        native_unit_of_measurement=None,
+        value_fn=lambda arve_data: getattr(arve_data, "tvoc"),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Arve device based on a config entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        [ArveDevice(coordinator, description) for description in SENSORS],
+        True,
+    )
+
+
+class ArveDevice(ArveDeviceEntity, SensorEntity):
+    """Define an Arve device."""
+
+    entity_description: ArveDeviceEntityDescription
+
+    def __init__(
+        self,
+        coordinator: ArveCoordinator,
+        description: ArveDeviceEntityDescription,
+    ) -> None:
+        """Initialize Arve device."""
+        super().__init__(coordinator)
+        self.coordinator = coordinator
+        self.sn = coordinator.arve.device_sn
+        self.formated_sn = "_".join(self.sn.lower().split("-"))
+        self.entity_description = description
+        self.trans_key = str(self.entity_description.translation_key)
+        self._attr_unique_id = "_".join(
+            [
+                self.sn,
+                self.trans_key,
+            ]
+        )
+
+        LOGGER.debug(self._attr_unique_id)
+
+        self.name = description.key
```

Comment:
does it make sense to move this (or at least parts of this) to the base class

Reply:
Moved part of this to the base class

Reply:
just go ahead and mark everything as resolved you think is addressed, so we can easier keep the overview.